ocaml/typing/env.ml

```diff
@@ -2677,13 +2677,31 @@ let save_signature_with_imports ~alerts sg modname filename imports =
   save_signature_with_transform with_imports
     ~alerts sg modname filename
 
-(* Make the initial environment *)
+(* Make the initial environment, without language extensions *)
 let (initial_safe_string, initial_unsafe_string) =
   Predef.build_initial_env
     (add_type ~check:false)
     (add_extension ~check:false ~rebind:false)
     empty
 
+let add_language_extension_types env =
+  lazy
+    ((* CR ccasinghino for mslater: Here, check the simd extension.  If it's on,
+        return [add_simd_extension_types (add_type ~check:false) env].
+        Otherwise, return env. *)
+      env)
+
+(* Some predefined types are part of language extensions, and we don't want to
+   make them available in the initial environment if those extensions are not
+   turned on.  We can't do this at startup because command line flags haven't
+   been parsed yet. So, we make the initial environment lazy.
+
+   It is important that [initial_safe_string] and [initial_unsafe_string] are
```

Comment:
"important": I think you mean that if the command arguments haven't yet been parsed, the env will be missing any extensions. For most usages, that's probably fine. I'm worried about cases where a compiler-libs call accesses the initial env (as is done in the tests) and there are no command arguments to speak of.

Reply:
Adjusted - see if you like it.